File: EPSILON/SAMPLE FILE/E-publishing/Tools/Altova/Altova/XMLSpy2006/spl/cs/AltovaText/TextNode.cs
[call GenerateFileHeader("TextNode.cs")]
namespace Altova.TextParser
{
	/// <summary>
	/// Encapsulates a text node in a tree as needed by parsing text input files.
	/// </summary>
	public class TextNode : ITextNode
	{
		#region Implementation Detail:
		ITextNode mParent = NullTextNode.Instance;
		string mName = "";
		string mValue = "";
		ITextNodeCollection mChildren = null;
		NodeClass mClass = NodeClass.Undefined;
		bool mHasDecimalData = false;
		uint mMaximumLength = 0;
		string mNativeName = "";
		string mPrecedingSeparators= "";
		string mFollowingSeparators= "";
		uint mPositionInFather= 0;
		#endregion
		#region Public Interface:
		/// <summary>
		/// Constructs an instance of this class.
		/// </summary>
		/// <param name="parent">the parent of this node</param>
		/// <param name="name">the name this instance should have</param>
		/// <remarks>
		/// Note that this instance will be automatically added to the children of the specified parent
		/// unless it's null.
		/// </remarks>
		public TextNode(ITextNode parent, string name): this(parent, name, NodeClass.Undefined)
		{
		}
		/// <summary>
		/// Constructs an instance of this class.
		/// </summary>
		/// <param name="parent">the parent of this node</param>
		/// <param name="name">the name this instance should have</param>
		/// <param name="eClass">the name of the node's class</param>
		/// <remarks>
		/// Note that this instance will be automatically added to the children of the specified parent
		/// unless it's null.
		/// </remarks>
		public TextNode(ITextNode parent, string name, NodeClass eClass)
		{
			mName= name;
			mChildren = new TextNodeCollection(this);
			mParent = parent;
			mParent.Children.Add(this);
			mClass = eClass;
		}
		/// <summary>
		/// Gets the root node of the hierarchy where this instance resides.
		/// If this instance has no parent (<see cref="Parent"/>==null),
		/// it's its own root node.
		/// </summary>
		public virtual ITextNode Root
		{
			get
			{
				if (mParent.IsNull) return this;
				else return mParent.Root;
			}
		}
		/// <summary>
		/// Gets the parent node of this instance. May return null if the node has no parent.
		/// </summary>
		public virtual ITextNode Parent
		{
			get
			{
				return mParent;
			}
			set
			{
				mParent = value;
			}
		}
		/// <summary>
		/// Searches for a node with the specfied name moving upwards in the tree.
		/// </summary>
		/// <param name="name">the name of the node to search for</param>
		/// <returns>the first node matching the specified name or null if no match was found</returns>
		public virtual ITextNode SearchUpwardsByName(string name)
		{
			if (name == mName) return this;
			return mParent.SearchUpwardsByName(name);
		}
		/// <summary>
		/// Gets the children of this instance.
		/// </summary>
		public virtual ITextNodeCollection Children
		{
			get
			{
				return mChildren;
			}
		}
		/// <summary>
		/// Get/sets the name of this instance.
		/// </summary>
		public virtual string Name
		{
			get
			{
				return mName;
			}
			set
			{
				mName = value;
			}
		}
		/// <summary>
		/// Get/sets the value of this instance.
		/// </summary>
		public virtual string Value
		{
			get
			{
				return mValue;
			}
			set
			{
				mValue = value;
			}
		}

		/// <summary>
		/// Gets whether this instance is a null object.
		/// </summary>
		public bool IsNull
		{
			get
			{
				return false;
			}
		}
		/// <summary>
		/// Get/sets the kind of this instance.
		/// </summary>
		public NodeClass Class
		{
			get
			{
				return mClass;
			}
			set
			{
				mClass = value;
			}
		}
		/// <summary>
		/// Get/sets whether this instance contains decimal data. Only makes sense if <see cref="ITextNode.Class"/> 
		/// returns <see cref="NodeClass.DataElement"/>.
		/// </summary>
		public bool HasDecimalData
		{
			get
			{
				return mHasDecimalData;
			}
			set
			{
				mHasDecimalData = value;
			}
		}
		/// <summary>
		/// Get/sets the maximum length the <see cref="Value"/> of this instance may have.
		/// </summary>
		public uint MaximumLength
		{
			get
			{
				return mMaximumLength;
			}
			set
			{
				mMaximumLength = value;
			}
		}
		/// <summary>
		/// Get/sets the native name of this instance.
		/// </summary>
		public string NativeName
		{
			get
			{
				return mNativeName;
			}
			set
			{
				mNativeName = value;
			}
		}
		/// <summary>
		/// Get/sets the separators that potentially could precede this instance.
		/// </summary>
		public string PrecedingSeparators
		{
			get
			{
				return mPrecedingSeparators;
			}
			set
			{
				mPrecedingSeparators= value;
			}
		}
		/// <summary>
		/// Get/sets the separators that potentially could follow this instance.
		/// </summary>
		public string FollowingSeparators
		{
			get
			{
				return mFollowingSeparators;
			}
			set
			{
				mFollowingSeparators= value;
			}
		}
		/// <summary>
		/// Get/sets the position of this instance in its father's children.
		/// </summary>
		public uint PositionInFather
		{
			get
			{
				return mPositionInFather;
			}
			set
			{
				mPositionInFather= value;
			}
		}
		#endregion
	}

}

File: EPSILON/SAMPLE FILE/E-publishing/Tools/Altova/Altova/XMLSpy2006/spl/java/altova/text/TextNode.java
[call GenerateFileHeader("TextNode.java")]
package com.altova.text;

public class TextNode implements ITextNode {
	private ITextNode m_Parent = NullTextNode.getInstance();

	private TextNodeList m_Children = null;

	private String m_Name = "";

	private String m_Value = "";

	private byte m_NodeClass = ITextNode.Undefined;

	private boolean m_HasDecimalData = false;

	private int m_MaximumLength = 0;

	private String m_NativeName = "";

	private String m_PrecedingSeparators = "";

	private String m_FollowingSeparators = "";

	private int m_PositionInFather = 0;

	public TextNode(ITextNode parent, String name) {
		this(parent, name, ITextNode.Undefined);
	}

	public TextNode(ITextNode parent, String name, byte nodeClass) {
		m_Name= name;
		m_NodeClass = nodeClass;
		m_Children = new TextNodeList(this);
		m_Parent = parent;
		m_Parent.getChildren().add(this);
	}

	public ITextNode getRoot() {
		return m_Parent.getRoot();
	}

	public ITextNode getParent() {
		return m_Parent;
	}

	public void setParent(ITextNode rhs) {
		m_Parent = rhs;
	}

	public ITextNodeList getChildren() {
		return m_Children;
	}

	public ITextNode findNodeUpwardsByName(String name) {
		if (name == m_Name)
			return this;
		return m_Parent.findNodeUpwardsByName(name);
	}

	public String getName() {
		return m_Name;
	}

	public void setName(String name) {
		m_Name = name;
	}

	public String getValue() {
		return m_Value;
	}

	public void setValue(String value) {
		m_Value = value;
	}

	public boolean isNull() {
		return false;
	}

	public byte getNodeClass() {
		return m_NodeClass;
	}

	public void setNodeClass(byte rhs) {
		m_NodeClass = rhs;
	}

	public boolean hasDecimalData() {
		return m_HasDecimalData;
	}

	public void setHasDecimalData(boolean rhs) {
		m_HasDecimalData = rhs;
	}

	public int getMaximumLength() {
		return m_MaximumLength;
	}

	public void setMaximumLength(int rhs) {
		m_MaximumLength = rhs;
	}

	public String getNativeName() {
		return m_NativeName;
	}

	public void setNativeName(String rhs) {
		m_NativeName = rhs;
	}

	public String getPrecedingSeparators() {
		return m_PrecedingSeparators;
	}

	public void setPrecedingSeparators(String rhs) {
		m_PrecedingSeparators = rhs;
	}

	public String getFollowingSeparators() {
		return m_FollowingSeparators;
	}

	public void setFollowingSeparators(String rhs) {
		m_FollowingSeparators = rhs;
	}

	public int getPositionInFather() {
		return m_PositionInFather;
	}

	public void setPositionInFather(int rhs) {
		m_PositionInFather = rhs;
	}

}
